Source/Scene/Scene.js

```diff
@@ -2696,6 +2706,68 @@ define([
         return object;
     };
 
+    var scratchPickDepthPassState;
+
+    function renderForPickDepth(scene, drawingBufferPosition) {
+        var context = scene._context;
+        var us = context.uniformState;
+        var frameState = scene._frameState;
+
+        // Update with previous frame's number and time, assuming that render is called before picking.
+        updateFrameState(scene, frameState.frameNumber, frameState.time);
+        frameState.cullingVolume = getPickCullingVolume(scene, drawingBufferPosition, 1, 1);
+        frameState.passes.render = true;
+
+        us.update(frameState);
+
+        var passState = scratchPickDepthPassState;
+        if (!defined(passState)) {
+            passState = scratchPickDepthPassState = new PassState(context);
```

Comment:
What if the app has more than one context?  Should `scratchPickDepthPassState` be part of the `Scene`?